### PR TITLE
update file upload api

### PR DESCRIPTION
**Pull request by dcmcand**
**Description of change**
Update file upload api in response to comments on https://github.com/HHS/Head-Start-TTADP/pull/267

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/pull/267

### CODE REVIEW

File: src/lib/s3Uploader.js [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
-  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
-  endpoint: process.env.S3_ENDPOINT,
-  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
-  signatureVersion: 'v4',
-  s3ForcePathStyle: true,
-});
+let s3Config;
+
+if (process.env.VCAP_SERVICES) {
+  const bucket = process.env.VCAP_SERVICES.s3[0];
```
Comment (rahearn) [inline src/lib/s3Uploader.js:6]: `VCAP_SERVICES` is a json string in the environment. I know in ruby you need to call `JSON.parse(ENV["VCAP_SERVICES"])` to get the values out. Does node handle that transparently?

File: src/lib/s3Uploader.js [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
+  s3Config = {
+    accessKeyId: bucket.access_key_id,
+    endpoint: bucket.uri,
+    secretAccessKey: bucket.secret_access_key,
```
Comment (rahearn) [inline src/lib/s3Uploader.js:10]: I think these paths won't quite be correct. See the example data I DM'd you in slack. There's a `credentials` object that they're held in inside of what you've assigned to `bucket`

File: src/lib/s3Uploader.js [hunk 3]
```diff
@@ -1,12 +1,26 @@
...
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+    endpoint: process.env.S3_ENDPOINT,
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    signatureVersion: 'v4',
+    s3ForcePathStyle: true,
+  };
+}
+export const s3 = new S3(s3Config);
 
 export const verifyVersioning = async (bucket = process.env.bucket, s3Client = s3) => {
```
Comment (rahearn) [inline src/lib/s3Uploader.js:25]: Setting bucket name should also come from `VCAP_SERVICES` in deployed environments. Looks like there's two points the bucket name is set, and right now they're being set to different env vars. `process.env.bucket` here and `process.env.S3_BUCKET` on line 47

**Review by rahearn — CHANGES_REQUESTED** [review]
Cleanup is all awesome, but I think the VCAP_SERVICES part needs a little more work.

File: src/lib/s3Uploader.js [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
-  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
-  signatureVersion: 'v4',
-  s3ForcePathStyle: true,
-});
+let s3Config;
 
-export const verifyVersioning = async (bucket = process.env.bucket, s3Client = s3) => {
+let bucketName = process.env.S3_BUCKET;
+if (process.env.VCAP_SERVICES) {
+  const { credentials, name } = process.env.VCAP_SERVICES.s3[0];
```
Comment (rahearn) [inline src/lib/s3Uploader.js:7]: Just verified this on cloud.gov using the node REPL. We need to parse the JSON string explicitly

```suggestion
  const { credentials, name } = JSON.parse(process.env.VCAP_SERVICES).s3[0];
```

Or to cleanup the next steps:

```suggestion
  const { s3: [ { credentials: { access_key_id, fips_endpoint, secret_access_key, bucket } } ] } = JSON.parse(process.env.VCAP_SERVICES);
```

**Review by rahearn — CHANGES_REQUESTED** [review]

File: src/lib/s3Uploader.js [hunk 5]
```diff
@@ -1,14 +1,30 @@
...
+  bucketName = name;
+  s3Config = {
+    accessKeyId: credentials.access_key_id,
+    endpoint: credentials.fips_endpoint,
+    secretAccessKey: credentials.secret_access_key,
+    signatureVersion: 'v4',
+    s3ForcePathStyle: true,
+  };
```
Comment (dcmcand) [inline src/lib/s3Uploader.js:15]: I tested all these paths with the sample data you sent and they all work now.

**Review by rahearn — APPROVED** [review]
🎉

**Pull request merged**